### PR TITLE
pyspark 3.5.3

### PR DESCRIPTION
**Pull request by lorepirri**
pyspark 3.5.3

**Destination channel:** defaults

### Links

- [PKG-6274]
- dev_url (pypi): https://github.com/apache/spark/tree/v3.5.3/python
- conda-forge: https://github.com/conda-forge/pyspark-feedstock/blob/main/recipe/meta.yaml
- pypi: https://pypi.org/project/pyspark/3.5.3
- pypi inspector: https://inspector.pypi.io/project/pyspark/3.5.3

### Explanation of changes:

- bump version

[PKG-6274]: https://anaconda.atlassian.net/browse/PKG-6274?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml
Comment (ViridianMelody) [inline recipe/meta.yaml:41]:
```suggestion
  run_constrained:
    - grpcio >=1.56.0
    - grpcio-status >=1.56.0
    - googleapis-common-protos >=1.56.4
```
We haven't previously included the "connect" extras, so we can keep these in run_constrained.

**Review by ViridianMelody — CHANGES_REQUESTED** [review]

**Review by ViridianMelody — APPROVED** [review]

File: recipe/meta.yaml
Comment (lorepirri) [inline recipe/meta.yaml]:
```suggestion
    # extras_require dependencies are kept here
    # as runtime dependencies because previously
    # were set as such
```

File: recipe/meta.yaml
Comment (lorepirri) [inline recipe/meta.yaml:39]:
```suggestion
    # We haven't previously included the "connect" extras, 
    # so we can keep these in run_constrained.
    - grpcio >=1.56.0
```

File: recipe/meta.yaml
Comment (lorepirri) [inline recipe/meta.yaml:56]:
```suggestion
    - pyspark.sql
    # imports relative to connect are removed because is an extra
```

**Review by ifitchet — APPROVED** [review]
per-Slack: some comments to avoid repeated questions on choices.

Given that you're removing all the CI constructs, maybe we need to wait for @dbast's thoughts.  Someone might be relying on them.

**Comment by lorepirri** [comment]
> per-Slack: some comments to avoid repeated questions on choices.
> 
> Given that you're removing all the CI constructs, maybe we need to wait for @dbast's thoughts. Someone might be relying on them.

I restored the CODEOWNERS, I think the CI is left over from conda-forge. I will wait for today for @dbast feedback on it.

**Comment by dbast** [comment]
Feel free to merge this without my review. I only get pinged as I am part of the AnacondaRecipes Github Org and I am in the CODEOWNERS file due to being maintainer of the recipe in the upstream feedstock. The PR looks fine, the CI files can be wiped as the anaconda build system only cares about the recipe + local {cbc, abs}.yml files.

**Comment by lorepirri** [comment]
> Feel free to merge this without my review. I only get pinged as I am part of the AnacondaRecipes Github Org and I am in the CODEOWNERS file due to being maintainer of the recipe in the upstream feedstock. The PR looks fine, the CI files can be wiped as the anaconda build system only cares about the recipe + local {cbc, abs}.yml files.

Thank you @dbast ! Will merge

**Pull request merged**
